server/plugin/virtual_agent.go

```diff
@@ -229,12 +240,69 @@ func (p *Plugin) ProcessResponse(data []byte) error {
 			if _, err = p.DMWithAttachments(userID, p.CreateOutputCardAttachment(&data)); err != nil {
 				return err
 			}
+		case *OutputImage:
+			post, err := p.CreateOutputImagePost(res, userID)
+			if err != nil {
+				return err
+			}
+
+			_, err = p.dm(userID, post)
+			if err != nil {
+				return err
+			}
 		}
 	}
 
 	return nil
 }
 
+func (p *Plugin) CreateOutputImagePost(body *OutputImage, userID string) (*model.Post, error) {
+	channel, appErr := p.API.GetDirectChannel(userID, p.botUserID)
+	if appErr != nil {
+		p.API.LogInfo("Couldn't get bot's DM channel", "user_id", userID, "error", appErr.Message)
+		return nil, appErr
+	}
+
+	resp, err := http.Get(body.Value)
+	if err != nil {
+		p.API.LogInfo("Error in getting file data from link", "error", err.Error())
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		p.API.LogInfo("Error in reading file data", "error", err.Error())
+		return nil, err
+	}
+
+	linkContents := strings.Split(body.Value, "/")
+	completeFilename := linkContents[len(linkContents)-1]
```

Comment:
here if len(linkContents) is 0 then 0-1=-1 and this will give a panick so have a check for its length